com.reprezen.swagedit.core/src/com/reprezen/swagedit/core/json/references/JsonReferenceValidator.java

```diff
@@ -84,47 +85,36 @@ public JsonReferenceValidator(JsonReferenceFactory factory) {
      * @param errors
      *            current set of errors
      */
-    private void validateType(JsonDocument doc, URI baseURI, AbstractNode node, JsonReference reference,
+    protected void validateType(JsonDocument doc, URI baseURI, AbstractNode node, JsonReference reference,
             Set<SwaggerError> errors) {
-        if (node == null) {
-            return;
-        }
 
-        Model model = doc.getModel();
+        AbstractNode target = findTarget(doc, baseURI, reference);
         TypeDefinition type = node.getType();
+        boolean isValidType = type != null && type.validate(target);
 
-        AbstractNode nodeValue = node.get(JsonReference.PROPERTY);
-        String pointer = (String) nodeValue.asValue().getValue();
-        AbstractNode valueNode = model.find(pointer);
+        if (!isValidType) {
+            errors.add(createReferenceError(SEVERITY_WARNING, error_invalid_reference_type, reference));
+        }
+    }
+
+    protected AbstractNode findTarget(JsonDocument doc, URI baseURI, JsonReference reference) {
```

Comment:
I understand why you used `target` here, but it's too generic. What about `referencedElement`?

Reply:
I kept target as it is shorter and meaning is same.